### PR TITLE
Team building changes for redesign

### PR DESCRIPTION
**Pull request by buoyad**
Prep work for integrating team building into the add to team wizard

- Make "Add/invite people" nav to step 1 of wizard
- Hook up "From keybase" to the team building modal
- Add `DefaultFullHeight` mode to `Modal`
  - Useful for cases like having a list within the modal
- Add a bunch of desktop style overrides to `Modal` to make team building still work in people search
- Switch team building to use modal instead of `HeaderHoc`/`PopupDialogHoc`

cc @keybase/y2ksquad

### CODE REVIEW

**Comment by buoyad** [comment]
@keybase/y2ksquad I ended up including the `teamsRedesign` changes in this pr.

File: shared/team-building/index.tsx [hunk 1]
```diff
@@ -569,6 +572,75 @@ class TeamBuilding extends React.PureComponent<Props> {
     ? Kb.ReAnimated.event([{nativeEvent: {contentOffset: {y: this.offset}}}], {useNativeDriver: true})
     : undefined
 
+  private modalHeader = () => {
```
Comment (buoyad) [inline shared/team-building/index.tsx]: Sorry if this function is kinda hard to follow :/

File: shared/team-building/index.tsx [hunk 2]
```diff
@@ -857,4 +918,11 @@ const styles = Styles.styleSheetCreate(
     } as const)
 )
 
+const peopleModalProps: Partial<React.ComponentProps<typeof Kb.Modal>> = {
```
Comment (jakob223) [inline shared/team-building/index.tsx]: are these styles something that would make sense to include in `DefaultFullHeight` or a rename of that mode? or do you expect they're one-off for this modal?

Comment (buoyad) [reply shared/team-building/index.tsx]: They're one-off for the people search version of this modal to make it appear over where the search bar is on that page.
